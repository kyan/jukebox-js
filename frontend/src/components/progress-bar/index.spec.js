import React from 'react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import { mount } from 'enzyme'
import ProgressBar from './index'

describe('ProgressBar', () => {
  const prevMock = jest.fn()
  const nextMock = jest.fn()
  const mockStore = configureMockStore()

  beforeEach(() => {
    jest.clearAllMocks()
  })

  describe('render', () => {
    it('renders the as expected', () => {
      const store = mockStore({
        timer: { duration: 10000, position: 8000, remaining: 700 },
        track: { length: 12345 }
      })
      const wrapper = mount(
        <Provider store={store}>
          <ProgressBar onPrevious={prevMock} onNext={nextMock} />
        </Provider>
      )
      expect(wrapper).toMatchSnapshot()
    })
  })
})
