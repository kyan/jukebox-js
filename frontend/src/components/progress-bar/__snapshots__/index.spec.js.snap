// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar render renders the as expected 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ProgressBar
    onNext={[MockFunction]}
    onPrevious={[MockFunction]}
  >
    <div
      className="c-progressBar"
    >
      <span
        className="c-progressBar__trackLength"
      >
        0:12
      </span>
      <span
        className="c-progressBar__remainingTime"
      >
        0:01
      </span>
      <Progress
        className=""
        percent={80}
        prefixCls="rc-progress"
        strokeColor="#2db7f5"
        strokeLinecap="square"
        strokeWidth={1}
        style={Object {}}
        trailColor="#D9D9D9"
        trailWidth={1}
      >
        <svg
          className="rc-progress-line "
          preserveAspectRatio="none"
          style={Object {}}
          viewBox="0 0 100 1"
        >
          <path
            className="rc-progress-line-trail"
            d="M 0,0.5
           L 100,0.5"
            fillOpacity="0"
            stroke="#D9D9D9"
            strokeLinecap="square"
            strokeWidth={1}
          />
          <path
            className="rc-progress-line-path"
            d="M 0,0.5
           L 100,0.5"
            fillOpacity="0"
            key="0"
            stroke="#2db7f5"
            strokeLinecap="square"
            strokeWidth={1}
            style={
              Object {
                "strokeDasharray": "80px, 100px",
                "strokeDashoffset": "-0px",
                "transition": "stroke-dashoffset 0.3s ease 0s, stroke-dasharray .3s ease 0s, stroke 0.3s linear",
              }
            }
          />
        </svg>
      </Progress>
    </div>
  </ProgressBar>
</Provider>
`;
